src/fenicsxconcrete/finite_element_problem/base_material.py

```diff
@@ -54,10 +103,8 @@ def __init__(
         self.pv_output_file = Path(pv_path) / (pv_name + ".xdmf")
 
         # setup fields for sensor output, can be defined in model
-        self.displacement = None
-        self.temperature = None
-        self.degree_of_hydration = None
-        self.q_degree_of_hydration = None
+        self.fields = None
```

Comment:
I would add here the two field classes in the comment

Reply:
Since (maybe) not all problems have a displacement field, or have more than one field (ThermoMechanicalModel), having a default `self.V` would again be ambiguous. I would suggest, that the reaction force sensor should just get the function space from the displacement field

```python
V = problem.fields.displacement.function_space
```